front/pages/api/lookup.ts

```diff
@@ -0,0 +1,192 @@
+  }
+
+  if (bearerTokenRes.value !== config.getLookUpBearerToken()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "invalid_basic_authorization_error",
+        message: "Invalid token",
+      },
+    });
+  }
+  const { resource } = req.query;
+  if (!resource || typeof resource !== "string") {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Missing or invalid resource parameter",
+      },
+    });
+  }
+
+  const resourceValidation = ResourceType.decode(resource);
+  if (isLeft(resourceValidation)) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid resource type. Must be 'user' or 'workspace'",
+      },
+    });
+  }
+
+  let response: LookupResponseBody | null = null;
+  switch (resourceValidation.right) {
+    case "user":
+      {
+        const bodyValidation = UserLookupSchema.decode(req.body);
+        if (isLeft(bodyValidation)) {
+          const pathError = reporter.formatValidationErrors(
+            bodyValidation.left
+          );
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: `Invalid request body for user lookup: ${pathError}`,
+            },
+          });
+        }
+        response = await handleLookupUser(bodyValidation.right);
+        const session = await getAuth0Session(req, res);
+        const sessionWithUser = {
+          user: bodyValidation.right.user,
+          session,
+        };
+        response = await handleLookupUser(sessionWithUser);
+      }
+      break;
+    case "workspace": {
+      const bodyValidation = WorkspaceLookupSchema.decode(req.body);
+      if (isLeft(bodyValidation)) {
+        const pathError = reporter.formatValidationErrors(bodyValidation.left);
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: `Invalid request body for user lookup ${pathError}`,
+          },
+        });
+      }
+      response = await handleLookupWorkspace(bodyValidation.right);
+    }
+  }
+  res.status(200).json(response);
+  return;
+}
+
+async function handleLookupUser(
+  body: SessionWithUser
+): Promise<UserLookupResponse> {
+  // Check if user exists
+  const user = await fetchUserFromSession(body);
+
+  // Check for pending invitations
+  const pendingInvite =
+    await getPendingMembershipInvitationWithWorkspaceForEmail(body.user.email);
+
+  // Check for workspace with verified domain
+  const workspaceWithVerifiedDomain =
+    await findWorkspaceWithVerifiedDomain(body);
+
+  // Check auto-join
+  const canAutoJoin =
+    workspaceWithVerifiedDomain?.domainAutoJoinEnabled ?? false;
+
+  return {
+    isNew: !user,
+    hasInvite: !!pendingInvite,
```

Comment:
nit: since isNew can't be true and hasInvite be true, I tends to prefer something like:
`status: unknown | invited | member`